### PR TITLE
fix: re-introduce CORS, at least for local_mode

### PR DESCRIPTION
**Pull request by tarilabs**
# Pull Request

## Description

in previous iterations, we had CORS for local mode; this re-introduces CORS headers as middleware handler, in this iteration as a `*` and only when `LOCAL_MODE=1`

## Type of Change

Please mark the relevant option(s):

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Test improvements
- [ ] CI/CD improvements
- [ ] Other (please describe):

## Component(s) Affected

Please mark the relevant component(s):

- [x] API endpoints
- [ ] Evaluation executors
- [ ] MLFlow integration
- [ ] Collection management
- [ ] Kubernetes/OpenShift deployment
- [ ] Configuration system
- [ ] Monitoring/metrics
- [ ] Documentation
- [ ] Tests
- [ ] CI/CD
- [ ] Other:

## Changes Made

as description

## Testing

Please describe how you tested your changes:

### Test Coverage
- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed
- [ ] No tests needed (documentation/minor changes)

### Test Results
- [x] All existing tests pass
- [x] New tests pass
- [x] Manual testing successful

**Manual Testing Details** (if applicable):

`LOCAL_MODE=1 make start-service`

image

https://editor.swagger.io/?url=https://raw.githubusercontent.com/eval-hub/eval-hub/refs/heads/main/api/openapi.yaml

### Performance Impact
- [x] No performance impact
- [ ] Performance improvement
- [ ] Performance regression (justified and documented)
- [ ] Performance impact unknown

## Documentation

- [x] Code comments added/updated
- [ ] API documentation updated
- [ ] README updated
- [ ] CONTRIBUTING guide updated
- [ ] Examples/tutorials added/updated
- [ ] No documentation changes needed

## Breaking Changes

none

## Security Considerations

- [x] No security implications <-- only when `LOCAL_MODE=1`
- [ ] Security review needed
- [ ] Authentication/authorization changes
- [ ] Sensitive data handling changes
- [ ] Network security changes

Please describe any security implications:

## Deployment Considerations

- [x] No deployment changes needed
- [ ] Environment variables added/changed
- [ ] Configuration changes required
- [ ] Database migrations needed
- [ ] Kubernetes manifests updated
- [ ] New dependencies added

**Deployment Notes**:

## Screenshots/Evidence

If applicable, add screenshots or other evidence of the changes:

## Checklist

Please confirm the following before submitting:

### Code Quality
- [ ] Code follows the project's style guidelines
- [ ] Self-review of code has been performed
- [ ] Code is properly commented
- [ ] No new linting warnings introduced

### Testing
- [ ] Tests have been added for new functionality
- [ ] All tests pass locally
- [ ] Test coverage is adequate
- [ ] No regression in existing functionality

### Documentation
- [ ] Documentation has been updated as needed
- [ ] API documentation reflects changes
- [ ] Examples updated if needed
- [ ] Breaking changes documented

### Review Readiness
- [ ] PR title is clear and descriptive
- [ ] PR description is complete
- [ ] Commit messages follow conventional commits format
- [ ] Branch is up to date with main
- [ ] Ready for review

## Additional Notes

Add any additional notes, context, or questions for reviewers:

---

**For Reviewers**: Please check that all automated checks pass before approving. Pay special attention to:
- Test coverage and quality
- API contract compatibility
- Security implications
- Performance impact
- Documentation completeness

## Summary by CodeRabbit

* **New Features**
  * CORS support added and automatically enabled in local development mode; sets standard CORS headers and handles OPTIONS preflight requests.

* **Tests**
  * Comprehensive unit tests validating CORS header values, preflight responses, pass-through behavior, header preservation and multiple edge cases.

* **Chores**
  * Startup script updated to invoke the server in local mode by default for local development.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by tarilabs** [comment]
image

https://editor.swagger.io/?url=https://raw.githubusercontent.com/eval-hub/eval-hub/refs/heads/main/docs/openapi.yaml

File: scripts/start_server.sh [hunk 1]
```diff
@@ -14,15 +14,22 @@ PORT="$4"
 
 export GOCOVERDIR="$5"
 
+# Check if LOCAL_MODE environment variable is set to enable local mode (CORS)
+EXTRA_FLAGS=""
+if [[ -n "${LOCAL_MODE}" ]]; then
+  EXTRA_FLAGS="-local"
+  echo "Env variable LOCAL_MODE detected: starting in LOCAL MODE (CORS * enabled)"
+fi
+
```
Comment (julpayne) [inline scripts/start_server.sh]: @tarilabs do we really need this because the script in only used to run up the server in local mode? If you really want to have an option can we at least make local mode enabled if no environment variable is ste?

**Review by julpayne — COMMENTED** [review]
Please see my comment?

**Comment by tarilabs** [comment]
re-demo:
image

https://editor.swagger.io/?url=https://raw.githubusercontent.com/eval-hub/eval-hub/refs/heads/main/docs/openapi.yaml
